packages/ubuntu_desktop_installer/lib/pages/who_are_you_page.dart

```diff
@@ -0,0 +1,328 @@
+import 'package:encrypt/encrypt.dart' as encrypt;
+import 'package:flutter/material.dart';
+import 'package:form_field_validator/form_field_validator.dart';
+import 'package:provider/provider.dart';
+import 'package:subiquity_client/subiquity_client.dart';
+import 'package:yaru/yaru.dart' as yaru;
+
+import '../routes.dart';
+import '../widgets/localized_view.dart';
+import '../widgets/validated_input.dart';
+import 'wizard_page.dart';
+
+/// The installer page for setting up the user data.
+///
+/// It uses [WizardPage] and [WizardAction] to create an installer page.
+class WhoAreYouPage extends StatefulWidget {
+  /// Creates a the installer page for setting up the user data.
+  const WhoAreYouPage({Key? key}) : super(key: key);
+
+  @override
+  _WhoAreYouPageState createState() => _WhoAreYouPageState();
+}
+
+/// An enum for storing the login strategy.
+enum LoginStrategy {
+  /// If selected can be used for the representation
+  /// of the user preference to always entering the password
+  /// at login.
+  REQUIRE_PASSWORD,
+
+  /// If selected can be used for the representation
+  /// of the user preference to log in without a password.
+  AUTO_LOGIN
+}
+
+class _WhoAreYouPageState extends State<WhoAreYouPage> {
+  late LoginStrategy _loginStrategy;
+  late TextEditingController _realNameController;
+  late TextEditingController _computerNameController;
+  late TextEditingController _usernameController;
+  late TextEditingController _passwordController;
+  late TextEditingController _confirmPasswordController;
+
+  final _usernameFormKey = GlobalKey<FormState>();
+  final _realNameFormKey = GlobalKey<FormState>();
+  final _computerNameFormKey = GlobalKey<FormState>();
+  final _passwordFormKey = GlobalKey<FormState>();
+  final _confirmPasswordFormKey = GlobalKey<FormState>();
+
+  @override
+  void initState() {
+    _realNameController = TextEditingController();
+    _computerNameController = TextEditingController();
+    _usernameController = TextEditingController();
+    _passwordController = TextEditingController();
+    _confirmPasswordController = TextEditingController();
+
+    _loginStrategy = LoginStrategy.REQUIRE_PASSWORD;
+    super.initState();
+  }
+
+  Widget findPassWordStrengthLabel(String value, String weakPasswordLabel,
```

Comment:
can we describe the strength as an enum, determine the strength in view model, and merely create a widget here?